src/Auth/AuthDirective.php

```diff
@@ -33,7 +33,7 @@ public static function definition(): string
     public function resolveField(FieldValue $fieldValue): callable
     {
         return function (): ?Authenticatable {
-            $guard = $this->directiveArgValue('guard', AuthServiceProvider::guard());
+            $guard = $this->directiveArgValue('guard', current(AuthServiceProvider::guards()) ?: null);
```

Comment:
What does `current()` do? What is the intent?

Reply:
Current is used here to get the first item of AuthServiceProvider::guards()

It fallback nicely to `false` on empty array. Whereas AuthServiceProvider::guards()[0] that would throw on empty array.

Unlink `@guard`, the `@auth` directive isn't meant to be mutli-guard currently in LH (should it ?) So I made choice to fallback on first configured guard. 

Other option would be, if no guard attribute is specified on `@auth`, to walk against all guards to find first authenticated. But I found this a bit strange that without attribute multiple guards would be supported, and with attribute only one guard is possible.

Let me now what do you prefer.

Reply:
I think we should also allow multiple guards here.

Reply:
Just to be sure I clearly understand your proposition,

So we change `@auth(guard: String)` and `@whereAuth(guard: String)` into something like
`@auth(with: [String!])` and `@whereAuth(with: [String!])` to be consistent with `@guard(with:)` ?

This will introduce much more breaking changes than I expected :)

Reply:
I was thinking `@auth(guards: [])`.

> This will introduce much more breaking changes than I expected :)

Reply:
Ok, I'll do this later this week-end, looks like this will involve a lot of changes, tests, documentation, ...
 
I'm glad you are so open to such API changes !

Reply:
Sweet, thanks! I believe cleaning up inconsistencies like this will be beneficial in the long run.